### PR TITLE
feat: automated service and team compliance scorecard with per-check drill-down

### LINKED ISSUE #554: feat: automated service and team compliance scorecard with per-check drill-down
### Problem Statement

Polaris collects governance data but provides no summary metric for how compliant a system or team actually is. Engineering managers and architects have no way to track governance posture or compare teams. There is no actionable "your team needs to address X" signal — only raw violation lists that require manual interpretation.

### Proposed Solution

Implement an on-demand scorecard computed from existing data for each `System` and `Team`. Checks:

1. **SBOM freshness** — `lastSbomScanAt` < 30 days
2. **No Eliminate violations** — zero TIME compliance violations
3. **No license violations** — zero license compliance violations
4. **No critical version constraint violations**
5. **TIME classification coverage** — all technologies used by the team/system have a TIME category assigned

New endpoints: `GET /api/systems/{name}/scorecard` and `GET /api/teams/{name}/scorecard` returning `{ score: N, maxScore: M, checks: [...] }`. Display on System and Team detail pages as a progress bar or letter grade with per-check drill-down. Compute at request time (no persistence required for v1).

### Alternatives Considered

Persist daily scorecard snapshots to enable trend charts — valuable long-term but not required for v1. User-defined scorecard checks in YAML — adds flexibility but increases complexity; start with hardcoded checks.

### Area

Backend/API

### Additional Context

Most valuable after the SBOM-to-compliance data flow fix is in place so violation counts accurately reflect reality. Can be implemented independently — results will simply be less meaningful if upstream data gaps exist. New service: `server/services/scorecard.service.ts`.

### PR DESCRIPTION
**Pull request by localgod**
## Description

Implements #554: an on-demand governance scorecard for Systems and Teams, computed at request time from existing compliance data (no persistence, per the issue's v1 scope). Each check reports pass/fail plus a drill-down detail message.

Closes #554

## Type of Change

- [x] New feature (non-breaking change that adds functionality)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Schema update (changes to database schemas)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Configuration or build changes

## Changes Made

- `GET /api/systems/{name}/scorecard` and `GET /api/teams/{name}/scorecard`, backed by a new `ScorecardService` computing 5 checks per entity: SBOM freshness (< 30 days), no Eliminate-classified technologies in use, no disallowed-license usage, no critical version-constraint violations, and TIME classification coverage of used technologies.
- New Cypher (`systems/scorecard.cypher`, `teams/scorecard.cypher`) and `getScorecardRaw()` repository methods for the raw signals; reuses `TeamRepository.findUsage()` and `VersionConstraintService.getViolations()` where equivalent logic already existed, so semver evaluation for critical constraints stays in one place.
- Bug fix found along the way: `VersionConstraintRepository.findViolations()` declared a `system` filter in `ViolationFilters` that was never wired into the query — added the missing `sys.name = $system` condition (the system scorecard's critical-constraint check depends on it).
- New `ComplianceScorecard.vue` component (progress bar + per-check list), wired into both the System and Team detail pages.
- New `Scorecard`/`ScorecardCheck`/`ScorecardCheckId` types in `types/api.d.ts`.
- Unit tests for the service (not-found, all-clean, all-failing, and edge cases like a team owning zero systems) and both new API handlers.
- Regenerated `public/openapi.json`.

## Screenshots

N/A (see verification notes below)

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my changes
- [x] I have commented my code where necessary
- [x] My changes generate no new warnings or errors
- [x] I have run `npm run lint` and fixed any issues
- [x] I have tested my changes locally with `npm run dev`
- [ ] I have tested the build with `npm run build`
- [x] I have updated documentation if needed

## Additional Notes

Verified against the live dev Neo4j database and the running dev server: hit both endpoints directly (including the 404 path) and confirmed the scorecard renders server-side on the System and Team detail pages with real data (e.g. `/systems/AzureMap`, `/teams/Platform%20Team`).

### CODE REVIEW

File: server/services/scorecard.service.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      this.versionConstraintService.getViolations({ team: name, severity: 'critical' })
+    ])
+
+    const checks: ScorecardCheck[] = [
+      this.teamSbomFreshnessCheck(raw.systemScans),
+      this.eliminateViolationsCheck(usage.summary.violations, usage.summary.totalTechnologies),
+      this.licenseViolationsCheck(raw.licenseViolationCount),
+      this.criticalConstraintCheck(criticalViolations.summary.critical),
+      this.classificationCoverageCheck(usage.summary.unapproved, usage.summary.totalTechnologies)
+    ]
```
[the remark this answers is not in the document]
Comment (localgod) [reply server/services/scorecard.service.ts:79]: Good catch — fixed in 7e4e2bf. `usage.summary.unapproved` only counts technologies with no `APPROVES` relationship at all; it misses `complianceStatus === 'unknown'` (an `APPROVES` relationship exists but its `time` isn't one of the recognized TIME values). Coverage now derives the unclassified count from `usage.usage`, counting both `'unapproved'` and `'unknown'` statuses. Added a dedicated test for the `'unknown'` case.

File: server/database/queries/systems/scorecard.cypher [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  UNWIND (CASE WHEN size(usedTechs) = 0 THEN [null] ELSE usedTechs END) AS tech
+  OPTIONAL MATCH (owner)-[envApproval:APPROVES]->(tech)
+    WHERE tech IS NOT NULL AND sys.environment IS NOT NULL AND envApproval.environment = sys.environment
+  OPTIONAL MATCH (owner)-[blanketApproval:APPROVES]->(tech)
+    WHERE tech IS NOT NULL AND blanketApproval.environment IS NULL
+  WITH tech, coalesce(envApproval.time, blanketApproval.time) AS resolvedTime
+  WHERE tech IS NOT NULL
+  RETURN count(tech) AS usedTechnologyCount,
+         count(CASE WHEN resolvedTime IS NULL THEN 1 END) AS unclassifiedCount,
+         count(CASE WHEN resolvedTime = 'eliminate' THEN 1 END) AS eliminateCount
```
[the remark this answers is not in the document]
Comment (localgod) [reply server/database/queries/systems/scorecard.cypher]: Good catch — fixed in 7e4e2bf. Switched to `count(DISTINCT tech)` / `count(DISTINCT CASE ... THEN tech END)` so the counts stay correct even if a team ever ends up with more than one matching `APPROVES` relationship for the same (team, tech, environment) — nothing at the DB level prevents that in Neo4j Community Edition.

File: server/repositories/team.repository.ts [hunk 3]
```diff
@@ -372,6 +377,30 @@ export class TeamRepository extends BaseRepository {
...
+    const { records } = await this.executeQuery(query, { name })
+    const record = records[0]!
+    const systemScans = (record.get('systemScans') as Array<{ system: string; lastSbomScanAt: unknown }>) ?? []
+    return {
+      systemScans: systemScans.map(scan => ({
+        system: scan.system,
+        lastSbomScanAt: scan.lastSbomScanAt == null ? null : String(scan.lastSbomScanAt)
+      })),
+      licenseViolationCount: record.get('licenseViolationCount')?.toNumber() || 0
+    }
```
[the remark this answers is not in the document]
Comment (localgod) [reply server/repositories/team.repository.ts:401]: Looked into this one: `systemScans` comes from `RETURN [x IN raw WHERE x IS NOT NULL] AS systemScans` in teams/scorecard.cypher, which is a Cypher list comprehension — it always evaluates to a list (possibly empty), never null, regardless of the input. So the cast can't actually fail for this query today, and other repositories in this codebase (e.g. `version-constraint.repository.ts`'s `mapToConstraint`) rely on the same guarantee without an `Array.isArray` guard. Since this matches the existing convention and there's no reachable path that violates it, I'm leaving this one as-is rather than adding defensive checks for a shape the query can't produce — happy to revisit if the query ever changes to return something other than a list comprehension result.
